tendermint/src/lite/types.rs

```diff
@@ -107,46 +85,156 @@ pub trait Commit {
-        signed_voting_power * 3 > total_voting_power
+        signed_voting_power * self.denominator > total_voting_power * self.numerator
+    }
+}
+
+impl Default for TrustThresholdFraction {
+    fn default() -> Self {
+        Self::new(1, 3)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::lite::{TrustThreshold, TrustThresholdFraction};
+
+    #[test]
+    fn default_is_enough_power() {
+        let threshold = TrustThresholdFraction::default();
+
+        // 100% > 33%
+        assert!(threshold.is_enough_power(3, 3));
+
+        // 66% > 33%
+        assert!(threshold.is_enough_power(2, 3));
+
+        // 33% <= 33%
+        assert_eq!(threshold.is_enough_power(1, 3), false);
+
+        // 1% < 33%
+        assert_eq!(threshold.is_enough_power(1, 100), false);
     }
 }
 
 /// Requester can be used to request [`SignedHeader`]s and [`ValidatorSet`]s for a
 /// given height, e.g., by talking to a tendermint fullnode through RPC.
-pub trait Requester {
-    // TODO(Liamsi): consider putting this trait and the Store into a separate module / file...
-    type SignedHeader: SignedHeader;
-    type ValidatorSet: ValidatorSet;
-
-    /// Request the signed header at height h.
-    fn signed_header<H>(&self, h: H) -> Result<Self::SignedHeader, Error>
-    where
-        H: Into<Height>;
+pub trait Requester<C, H>
+where
+    C: Commit,
+    H: Header,
+{
+    /// Request the [`SignedHeader`] at height h.
+    fn signed_header(&self, h: Height) -> Result<SignedHeader<C, H>, Error>;
 
     /// Request the validator set at height h.
-    fn validator_set<H>(&self, h: H) -> Result<Self::ValidatorSet, Error>
-    where
-        H: Into<Height>;
+    fn validator_set(&self, h: Height) -> Result<C::ValidatorSet, Error>;
 }
 
 /// This store can be used to store all the headers that have passed basic verification
 /// and that are within the light client's trust period.
-pub trait Store {
-    type TrustedState: TrustedState;
-
+pub trait Store<C, H>
+where
+    H: Header,
+    C: Commit,
+{
     /// Add this state (header at height h, validators at height h+1) as trusted to the store.
-    // TODO: should this really take a reference?
-    fn add(&mut self, trusted: &Self::TrustedState) -> Result<(), Error>;
+    fn add(&mut self, trusted: TrustedState<C, H>) -> Result<(), Error>;
 
     /// Retrieve the trusted state at height h if it exists.
     /// If it does not exist return an error.
     /// If h=0, return the latest trusted state.
     /// TODO: use an enum instead of special-casing 0, see
     /// https://github.com/interchainio/tendermint-rs/issues/118
-    fn get(&self, h: Height) -> Result<&Self::TrustedState, Error>;
+    fn get(&self, h: Height) -> Result<&TrustedState<C, H>, Error>;
+}
+
+/// TrustedState stores the latest state trusted by a lite client,
+/// including the last header (at height h-1) and the validator set
+/// (at height h) to use to verify the next header.
+#[derive(Clone)]
+pub struct TrustedState<C, H>
```

Comment:
I do remember we've decided to make this (and `SignedHeader `) a struct. I couldn't find an issue where this decision was made. Hence I didn't link to it.